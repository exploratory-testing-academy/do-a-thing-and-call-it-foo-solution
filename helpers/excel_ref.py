import xlrd

class ExcelRef:

    def extract_cell_classic(number):
        loc = ("ref.xls")
        wb = xlrd.open_workbook(loc)
        sheet = wb.sheet_by_index(0)
        return sheet.cell_value(number, 1)

    def extract_cell_simplified(number):
        loc = ("ref.xls")
        wb = xlrd.open_workbook(loc)
        sheet = wb.sheet_by_index(0)
        return sheet.cell_value(number, 5)