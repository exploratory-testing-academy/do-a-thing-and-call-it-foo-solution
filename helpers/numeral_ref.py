from playwright.sync_api import Page

class NumeralRefPage:

    def __init__(self, page: Page):
        self.page = page

    def numeral_ref(self, inputnumber):
        url = "https://www.knowtheromans.com/roman-numerals/"
        self.page.goto(url)
        self.page.fill("#convertnumeralinput", str(inputnumber))
        outputnumber = self.page.inner_html("#numeralsout").strip("\n")
        return str(outputnumber)
