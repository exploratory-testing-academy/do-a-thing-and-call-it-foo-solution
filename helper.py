from helpers.numeral_ref import NumeralRefPage as r
from helpers.excel_ref import ExcelRef as e

def test_ref(browser_page):
    assert r(browser_page).numeral_ref(4) == 'IV'

def test_excel():
    assert e.extract_cell_classic(4) == 'IV'
    assert e.extract_cell_simplified(4) == 'IV'

def excel(num):
    return e.extract_cell_simplified(num)

def test_online_and_excel(browser_page):
    assert r(browser_page).numeral_ref(399) == e.extract_cell_classic(399)

def numbers_list(num):
    num_list = []
    for i in range(1, num + 1):
        num_list.append(i)
    return num_list

from approvaltests.combination_approvals import verify_all_combinations

def test_all_on_excel():
    verify_all_combinations(excel, [
        numbers_list(100)])

def test_all_on_browser(browser_page):
    verify_all_combinations(r(browser_page).numeral_ref, [
        numbers_list(100)])



# Developer intent, part 0. Review for correctness and consiceness

# Developer intent, part 1. Input -> Output
# Developer intent, part 2. Rules of behavior boundaries
# Developer intent, part 3. Coverage (pytest --cov=important_program important_program.py)
# Developer intent, part 4. Sampling vs wide nets (approvals)
# Developer intent, part 5. Properties

# Domain, laymen. Why oh why this?
# Domain, laymen. Rules. More rules.

# Domain, expert. Why oh why this?
# Domain, expert. Real and real boundaries.

# References. Give me the answers!
# References. Answers aren't as simple as you think.

# User filtering. No user would do what users would do.