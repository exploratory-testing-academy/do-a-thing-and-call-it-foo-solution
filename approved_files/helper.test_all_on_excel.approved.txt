args: (1,) => 'I'
args: (2,) => 'II'
args: (3,) => 'III'
args: (4,) => 'IV'
args: (5,) => 'V'
args: (6,) => 'VI'
args: (7,) => 'VII'
args: (8,) => 'VIII'
args: (9,) => 'IX'
args: (10,) => 'X'
args: (11,) => 'XI'
args: (12,) => 'XII'
args: (13,) => 'XIII'
args: (14,) => 'XIV'
args: (15,) => 'XV'
args: (16,) => 'XVI'
args: (17,) => 'XVII'
args: (18,) => 'XVIII'
args: (19,) => 'XIX'
args: (20,) => 'XX'
args: (21,) => 'XXI'
args: (22,) => 'XXII'
args: (23,) => 'XXIII'
args: (24,) => 'XXIV'
args: (25,) => 'XXV'
args: (26,) => 'XXVI'
args: (27,) => 'XXVII'
args: (28,) => 'XXVIII'
args: (29,) => 'XXIX'
args: (30,) => 'XXX'
args: (31,) => 'XXXI'
args: (32,) => 'XXXII'
args: (33,) => 'XXXIII'
args: (34,) => 'XXXIV'
args: (35,) => 'XXXV'
args: (36,) => 'XXXVI'
args: (37,) => 'XXXVII'
args: (38,) => 'XXXVIII'
args: (39,) => 'XXXIX'
args: (40,) => 'XL'
args: (41,) => 'XLI'
args: (42,) => 'XLII'
args: (43,) => 'XLIII'
args: (44,) => 'XLIV'
args: (45,) => 'VL'
args: (46,) => 'VLI'
args: (47,) => 'VLII'
args: (48,) => 'VLIII'
args: (49,) => 'IL'
args: (50,) => 'L'
args: (51,) => 'LI'
args: (52,) => 'LII'
args: (53,) => 'LIII'
args: (54,) => 'LIV'
args: (55,) => 'LV'
args: (56,) => 'LVI'
args: (57,) => 'LVII'
args: (58,) => 'LVIII'
args: (59,) => 'LIX'
args: (60,) => 'LX'
args: (61,) => 'LXI'
args: (62,) => 'LXII'
args: (63,) => 'LXIII'
args: (64,) => 'LXIV'
args: (65,) => 'LXV'
args: (66,) => 'LXVI'
args: (67,) => 'LXVII'
args: (68,) => 'LXVIII'
args: (69,) => 'LXIX'
args: (70,) => 'LXX'
args: (71,) => 'LXXI'
args: (72,) => 'LXXII'
args: (73,) => 'LXXIII'
args: (74,) => 'LXXIV'
args: (75,) => 'LXXV'
args: (76,) => 'LXXVI'
args: (77,) => 'LXXVII'
args: (78,) => 'LXXVIII'
args: (79,) => 'LXXIX'
args: (80,) => 'LXXX'
args: (81,) => 'LXXXI'
args: (82,) => 'LXXXII'
args: (83,) => 'LXXXIII'
args: (84,) => 'LXXXIV'
args: (85,) => 'LXXXV'
args: (86,) => 'LXXXVI'
args: (87,) => 'LXXXVII'
args: (88,) => 'LXXXVIII'
args: (89,) => 'LXXXIX'
args: (90,) => 'XC'
args: (91,) => 'XCI'
args: (92,) => 'XCII'
args: (93,) => 'XCIII'
args: (94,) => 'XCIV'
args: (95,) => 'VC'
args: (96,) => 'VCI'
args: (97,) => 'VCII'
args: (98,) => 'VCIII'
args: (99,) => 'IC'
args: (100,) => 'C'
